server/src/main/java/io/druid/metadata/SQLMetadataSupervisorManager.java

```diff
@@ -91,7 +92,7 @@ public Void withHandle(Handle handle) throws Exception
                     getSupervisorsTable()
                 )
             )
-                  .bind("id", String.format("%s_%s", id, version))
+                  .bind("id", String.format("%s_%s_%s", id, version, UUIDUtils.generateUuid()))
```

Comment:
@dclim haha, yes, it is. I wrote that most recent comment without seeing yours so it looks like we're on the same page :)